packages/crawler/src/evaluate-page.ts

```diff
@@ -0,0 +1,155 @@
+import { TagName } from 'shared/src';
+import { ConsumedTagNamesForVersionsAndPrefixes, TagNameData, TagNamesWithPropertyNames } from './types';
+
+declare global {
+  // eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+  interface Document {
+    // Extend Document interface, so we don't have to cast it on any
+    porscheDesignSystem: { [version: string]: { prefixes: string[] } };
+  }
+}
+
+export const evaluatePage = async (
+  page: puppeteer.Page,
+  pdsTagNamesWithPropertyNames: TagNamesWithPropertyNames
+): Promise<ConsumedTagNamesForVersionsAndPrefixes> => {
+  const pdsCrawlerRawData = await page.evaluate(
+    async ({ pdsTagNamesWithPropertyNames }): Promise<ConsumedTagNamesForVersionsAndPrefixes> => {
+      const tagNames = Object.keys(pdsTagNamesWithPropertyNames);
+
+      const getPdsComponentNameByPrefix = (el: Element, prefix: string): TagName | null => {
+        const tagName = el.tagName.toLowerCase();
+        return (
+          (Object.keys(pdsTagNamesWithPropertyNames).find(
+            (compName) => (prefix ? `${prefix}-${compName}` : compName) === tagName
+          ) as TagName) || null
+        );
+      };
+
+      const getAllChildElements = (el: Element): Element[] => {
+        const children = Array.from(el.children)
+          .concat(Array.from(el.shadowRoot?.children || []))
+          .flat() as Element[];
+        return children.concat(children.map(getAllChildElements).flat());
+      };
+
+      const getChildren = (el: Element): string | null =>
+        Array.from(el.children).reduce((result, slotEl) => {
+          if (slotEl.children.length) {
+            const copy = slotEl.cloneNode(true) as Element;
+            // we save only the highest dom level in the reports, in order not to make them too big
+            copy.innerHTML = '(the content was stripped)';
+            return result + copy.outerHTML;
+          }
+          return result + slotEl.outerHTML;
+        }, '' as string) || el.textContent;
+
+      const getHostPdsComponent = (el: Element, prefix: string): TagName | null => {
+        const rootNode = (el.getRootNode() as ShadowRoot).host;
+        if (rootNode) {
+          return getPdsComponentNameByPrefix(rootNode, prefix);
+        }
+        return null;
+      };
+
+      const parseIfJSON = (jsonStr: string | number | symbol): object | string | number | symbol => {
+        if (typeof jsonStr === 'string' && jsonStr.includes('{')) {
+          try {
+            // jsonStr is potentially JSON parsable string, e.g. "{ base: 'block', l: 'inline' }"
+            return JSON.parse(
+              jsonStr
+                .replace(/'/g, '"') // convert single quotes to double quotes
+                .replace(/[\s"]?([a-z\-]+)[\s"]?:([^//])/g, '"$1":$2') // wrap keys in double quotes if they don't have them but ignore potential urls
+            );
+          } catch {
+            // jsonStr is string, e.g. "block" or "inline"
+            return jsonStr;
```

Comment:
Probably a dead brach because of 2nd condition: `&& jsonStr.includes('{')`.

Reply:
It was covering the cases like "true" or "123", but actually it's better if we parse them too, so removed `&& jsonStr.includes('{')`